### PR TITLE
Add support for Unicode 10.0

### PR DESCRIPTION
**Pull request by sarowe**

### CODE REVIEW

File: jflex/src/main/java/jflex/core/unicode/UnicodeProperties.java [hunk 1]
```diff
@@ -19,19 +19,14 @@
...
- * downloaded from unicode.org.
- *
- * @author JFlex contributors.
- */
+// DO NOT EDIT
+// This class was automatically generated by jflex-unicode-maven-plugin
+// based on data files downloaded from unicode.org.
 public class UnicodeProperties {
 
-  /** Constant {@code UNICODE_VERSIONS="1.1, 1.1.5, 2, 2.0, 2.0.14, 2.1, 2.1.9,"{trunked}} */
```
Comment (sarowe) [inline jflex/src/main/java/jflex/core/unicode/UnicodeProperties.java]: @regisd : how are these javadoc comments generated?  I don't know how to put them back.

**Comment by sarowe** [comment]
TODO: Add examples/tests for Unicode segmentation rules (UAX#29 & UAX#14) under testsuite cases `unicode-{grapheme,sentence,word}-break/`, like for 8.0: `UnicodeGraphemeBreakRules_8_0`, `UnicodeSentenceBreakRules_8_0`, `UnicodeWordBreakRules_8_0`.  

Note that I had done this for all Unicode versions from 5.2->8.0, but didn't get to it for 9.0, so I plan on adding 9.0's along with 10.0's under this PR.

**Comment by sarowe** [comment]
This is ready to go.

I've included it in the 1.8 milestone, though, since I don't think adding Unicode version support belongs in a patch release, so I'm going to hold off merging until we've either released or decided to skip 1.7.1.
